### PR TITLE
[Impeller] add explainer for Android CPU profiling.

### PR DESCRIPTION
**Pull request by jonahwilliams**
So you don't have to talk to me to figure out how to do this.

### CODE REVIEW

File: impeller/docs/android_cpu_profile.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Add the `--no-stripped` flag to the gn config when building the android engine.
+
+Example config:
+
+`gn --no-lto --no-goma --runtime-mode=profile --android --android-cpu=arm64 --no-stripped`
+
+
+2. Configure Gradle to not remove strip sources
+
+In the flutter project file android/app/build.gradle, add the following line under the android block:
```
Comment (gaaclarke) [inline impeller/docs/android_cpu_profile.md:18]:
```suggestion
In the flutter project file `android/app/build.gradle`, add the following line under the `android` block:
```

File: impeller/docs/android_cpu_profile.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+```
+    packagingOptions{
+        doNotStrip "**/*.so"
+    }
+```
+
+3. `flutter run` the app with the local engine flags.
```
Comment (gaaclarke) [inline impeller/docs/android_cpu_profile.md:26]:
```suggestion
3. `flutter run` the app with the local engine flags (`--local-engine`, `--local-engine-host`, `--local-engine-src-path`)
```

File: impeller/docs/android_cpu_profile.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+4. Open Android Studio.
+
+You can create a new blank project if you don't have one already. You do not need to open the application project nor do you need to run the App through Android Studio. In fact, its much easier if you do not do those things.
```
Comment (gaaclarke) [inline impeller/docs/android_cpu_profile.md:29]:
```suggestion
You can create a new blank project if you don't have one already. You do not need to open the application project nor do you need to run the App through Android Studio. In fact, it's much easier if you do not do those things.
```

**Review by gaaclarke — APPROVED** [review]
LGTM I would prefer that this follows the same markdown guidelines we use for the webpage with respect to the column width (80 characters).  There are also linters for md in vscode but I didn't notice anything.

File: impeller/docs/android_cpu_profile.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+
+# Android CPU Profiling
+
+Android devices have different performance characteristics than iOS devices, and I've found that CPU traces frequently reveal surprising performance issues, such as https://github.com/flutter/engine/pull/48303 . This document describes the steps to capture an equivalent [flame graph](https://cacm.acm.org/magazines/2016/6/202665-the-flame-graph/abstract) on your local Android device.
```
Comment (matanlurey) [inline impeller/docs/android_cpu_profile.md:4]: Scratch "I've found that", as it's reviewed/in our repo it doesn't need an "I found".

**Review by matanlurey — APPROVED** [review]

**Comment by jonahwilliams** [comment]
I installed a markdown formatter and it actually removed all of the line breaks. I don't think we need to enforce 80 col for something that we're not reading the source for 🤷‍♂️

File: impeller/docs/android_cpu_profile.md (hunk 2 above)
Comment (jonahwilliams) [reply impeller/docs/android_cpu_profile.md:26]: Done

File: impeller/docs/android_cpu_profile.md (hunk 4 above)
Comment (jonahwilliams) [reply impeller/docs/android_cpu_profile.md]: Done

**Pull request merged**